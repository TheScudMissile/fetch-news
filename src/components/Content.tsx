import React, { useEffect, useState } from 'React';
import {
  Body,
  Button,
  Card,
  CardItem,
  Content,
  Left,
  Text,
  Toast,
} from 'native-base';
import { NEWS_API_KEY } from '../constants';
import { Article } from '../types';
import axios from 'axios';
import { Linking } from 'react-native';

const fetchTopHeadlines = async (
  setNewsArticles: (articles: Article[]) => void
) => {
  try {
    const response = await axios.get(
      `https://newsapi.org/v2/top-headlines?country=us&apiKey=${NEWS_API_KEY}`
    );
    setNewsArticles(response.data.articles);
  } catch (error) {
    console.error(error);
    Toast.show({
      buttonText: 'OK',
      duration: 3000,
      text: 'Failed to load articles.',
    });
  }
};

const navigateToArticle = (url: string) => {
  try {
    Linking.openURL(url);
  } catch (error) {
    console.error(error);
    Toast.show({
      buttonText: 'OK',
      duration: 3000,
      text: 'Unable to view article.',
    });
  }
};

export default () => {
  const [newsArticles, setNewsArticles] = useState<Article[]>([]);

  useEffect(() => {
    fetchTopHeadlines(setNewsArticles);
  }, []);

  return (
    <Content padder>
      {newsArticles.map(article => (
        <Card
          key={article.url}
          style={{ marginBottom: 15, marginLeft: 15, marginRight: 15 }}
        >
          <CardItem>
            <Left>
              <Body>
                <Text style={{ marginBottom: 15 }}>Title: {article.title}</Text>
                <Text style={{ marginBottom: 15 }}>
                  Description: {article.description}
                </Text>
                <Text style={{ marginBottom: 15 }}>
                  Published: {article.publishedAt}
                </Text>
                <Button
                  color="#34baeb"
                  transparent
                  style={{ marginLeft: 0 }}
                  onPress={() => navigateToArticle(article.url)}
                >
                  <Text style={{ paddingLeft: 0 }}>Full Article</Text>
                </Button>
              </Body>
            </Left>
          </CardItem>
        </Card>
      ))}
    </Content>
  );
};
