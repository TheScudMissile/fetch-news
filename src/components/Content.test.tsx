import * as React from 'react';
import { create } from 'react-test-renderer';
import Content from './Content';

describe('Content', () => {
  test('should render correctly', () => {
    const tree = create(<Content />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
