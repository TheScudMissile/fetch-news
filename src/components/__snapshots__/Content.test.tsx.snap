// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": 10,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  padder={true}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View />
</RCTScrollView>
`;
